### ISSUE TITLE
Incomplete types for installed package (httpx)

### ISSUE THREAD

**Issue by trendels** [microsoft/pyright#2473]
**Describe the bug**
Sorry for the vague description, I have so far not be able to narrow down this issue in any way, though it's easy to reproduce:

The [httpx](https://pypi.org/project/httpx/) package ships with type annotations. When I install the package in the current virtualenv using `pip install` and then create an instance of the `httpx.Response` class ([source](https://github.com/encode/httpx/blob/0.20.0/httpx/_models.py#L1197)) in a Python source file, pyright shows the type of the `status_code` property (and others) as `Unknown`, even though it is annotated as `int`. However, the type of properties implemented as `@property` (for example `http_version`) on the same class is detected correctly:

```python
from httpx import Response
response = Response(200)
reveal_type(response.http_version)
reveal_type(response.status_code)
```
Pyright output:
```
  /tmp/pyright_httpx/test_httpx.py:3:13 - info: Type of "response.http_version" is "str"
  /tmp/pyright_httpx/test_httpx.py:4:13 - info: Type of "response.status_code" is "Unknown"
```

In order to rule out problems with my local environment, I also tested without a virtualenv, with the extracted package source in the current directory (under `httpx`) In this case, *both* types are detected correctly:
```
  /tmp/pyright_httpx/test_httpx.py:3:13 - info: Type of "response.http_version" is "str"
  /tmp/pyright_httpx/test_httpx.py:4:13 - info: Type of "response.status_code" is "int"
```

**To Reproduce**
[Script to reproduce](https://gist.github.com/trendels/99116fd4d07e9e0a077cfeea96f84d45)

**Expected behavior**
Both types should be detected in both cases.

**Screenshots or Code**
Full output of script:
```
$ ./test.sh 
pyright 1.1.180

With installed package
======================
No configuration file found.
No pyproject.toml file found.
stubPath /tmp/pyright_httpx/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
/tmp/pyright_httpx/test_httpx.py
  /tmp/pyright_httpx/test_httpx.py:3:13 - info: Type of "response.http_version" is "str"
  /tmp/pyright_httpx/test_httpx.py:4:13 - info: Type of "response.status_code" is "Unknown"
0 errors, 0 warnings, 2 infos 
Completed in 0.898sec

With package source
===================
No configuration file found.
No pyproject.toml file found.
stubPath /tmp/pyright_httpx/httpx-0.20.0/typings is not a valid directory.
Assuming Python platform Linux
Searching for source files
Found 1 source file
/tmp/pyright_httpx/test_httpx.py
  /tmp/pyright_httpx/test_httpx.py:3:13 - info: Type of "response.http_version" is "str"
  /tmp/pyright_httpx/test_httpx.py:4:13 - info: Type of "response.status_code" is "int"
0 errors, 0 warnings, 2 infos 
Completed in 0.916sec
```

**VS Code extension or command-line**
Tested with pyright 1.1.80 from the command line.
